tests/test_bookmarks.py

```diff
@@ -90,34 +97,44 @@ def test_run(self):
-                        replication_key_value_parsed = parse(replication_key_value).strftime("%Y-%m-%dT%H:%M:%SZ")
+                        replication_key_value_parsed = self.dt_to_ts(replication_key_value, self.REPLICATION_DATE_FOMAT)
                         self.assertLessEqual(
                             replication_key_value_parsed, second_bookmark_value_parsed,
-                            msg="Second sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                            msg="Second sync bookmark was set incorrectly, \
+                                a record with a greater replication-key value was synced."
                         )
 
-                        # Verify the data of the second sync is greater-equal to the bookmark from the first sync
+                        # Verify the data of the second sync is greater-equal to the bookmark from
+                        # the first sync.
                         # We have added 'second_start_date' as the bookmark, it is more recent than
-                        #   the default start date and it will work as a simulated bookmark
+                        # the default start date and it will work as a simulated bookmark
                         self.assertGreaterEqual(
-                            replication_key_value_parsed, parse(self.second_start_date).strftime("%Y-%m-%dT%H:%M:%SZ"),
-                            msg="Sync did not respect the bookmark, a record with a smaller replication-key value was synced."
+                            replication_key_value_parsed, self.dt_to_ts(self.second_start_date, self.START_DATE_FORMAT),
+                            msg="Second sync did not respect the bookmark, \
+                                a record with a smaller replication-key value was synced."
                         )
 
                     for record in first_sync_messages:
 
-                        # Verify the first sync bookmark value is the max replication key value for a given stream
+                        # Verify the first sync bookmark value is the max replication key value for
+                        # a given stream
                         replication_key_value = record.get(replication_key)
-                        replication_key_value_parsed = parse(replication_key_value).strftime("%Y-%m-%dT%H:%M:%SZ")
+                        replication_key_value_parsed = self.dt_to_ts(replication_key_value, self.REPLICATION_DATE_FOMAT)
                         self.assertLessEqual(
                             replication_key_value_parsed, first_bookmark_value_parsed,
-                            msg="First sync bookmark was set incorrectly, a record with a greater replication-key value was synced."
+                            msg="First sync bookmark was set incorrectly, \
+                                a record with a greater replication-key value was synced."
                         )
 
                 else:
```

Comment:
instead of else block, can you keep this block of code for FULL_TABLE replication?

Reply:
@rdeshmukh15 un-clear about expectation over here. Can you please elaborate it?